src/index.ts

```diff
@@ -46,16 +46,77 @@ export default class DataSource {
       .join('&');
 
     const url = `${this.sorcererUrlBase}/${this.key}?${paramStr}`;
+    const isTokenBuilder = options['body'] && JSON.parse(options['body']).tokenApiVersion;
 
     options['cacheTime'] = options['cacheTime'] || 10 * 1000;
     options['headers'] = options['headers'] || {};
 
     options['headers']['x-reverse-proxy-ttl'] = options['cacheTime'] / 1000;
-    options['headers']['x-mi-cbe'] = CD._hashForRequest(url, options);
+    options['headers']['x-mi-cbe'] = isTokenBuilder
+      ? this.generateTokenBuilderHash(params, options)
```

Comment:
non-blocking: when Token Builder API is used the `params` argument passed to `getRawData` is always an empty object
```
//example from the docs
const tokens = [replaceToken, secretToken, hmacToken];
const requestBuilder = new RequestBuilder(tokens);
const postBody = JSON.stringify(requestBuilder.toJSON());

const options = {
   method: 'POST',
   body: postBody,
};

let { data } = await ds.getRawData({}, options);
```

So when we call `generateTokenBuilderHash` we don't have to pass it in.